src/main/java/fr/esgi/al5_2/Tayarim/services/ProprietaireService.java

```diff
@@ -16,141 +16,229 @@
+    numTel = numTel.replaceAll(" ", "");
+    if (proprietaireRepository.findFirstByNumTel(numTel).isPresent()) {
+      throw new ProprietaireNumTelAlreadyExistException();
     }
-
-    public List<ProprietaireDTO> getProprietaire(boolean isLogement){
-        return ProprietaireMapper.entityListToDtoList(proprietaireRepository.findAll(), isLogement);
+    proprietaireCreationDTO.setNumTel(numTel);
+
+    StringBuilder generatedPassword = new StringBuilder(16);
+    String allowedchar = "abcdefghijklmnopqrstuvwxyz0123456789!@#$%&*()_+-=[]?";
+    SecureRandom random = new SecureRandom();
+    for (int i = 0; i < 16; i++) {
+      int index = random.nextInt(allowedchar.length());
+      generatedPassword.append(allowedchar.charAt(index));
     }
 
-    public ProprietaireDTO getProprietaireById(@NonNull Long id, boolean isLogement){
-
-        Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(id);
-        if(optionalProprietaire.isEmpty()){
-            throw new ProprietaireNotFoundException();
-        }
-        return ProprietaireMapper.entityToDto(optionalProprietaire.get(), isLogement);
+    System.out.println(generatedPassword); // waiting for SMTP
+
+    Proprietaire proprietaire = ProprietaireMapper.creationDtoToEntity(proprietaireCreationDTO,
+        hashPassword(generatedPassword.toString()));
+    return ProprietaireMapper.entityToDto(proprietaireRepository.save(proprietaire), false);
+  }
+
+  /**
+   * Récupère une liste de tous les propriétaires.
+   *
+   * @param isLogement Indique si les logements associés doivent être inclus.
+   * @return Liste de DTOs des propriétaires.
+   */
+  public List<ProprietaireDTO> getProprietaire(boolean isLogement) {
+    return ProprietaireMapper.entityListToDtoList(proprietaireRepository.findAll(), isLogement);
+  }
+
+  /**
+   * Récupère un propriétaire par son identifiant, avec la possibilité d'inclure les logements
+   * associés.
+   *
+   * @param id         L'identifiant du propriétaire à récupérer.
+   * @param isLogement Indique si les logements associés doivent être inclus.
+   * @return Le DTO du propriétaire.
+   * @throws ProprietaireNotFoundException si le propriétaire n'est pas trouvé.
+   */
+  public ProprietaireDTO getProprietaireById(@NonNull Long id, boolean isLogement) {
+
+    Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(id);
+    if (optionalProprietaire.isEmpty()) {
+      throw new ProprietaireNotFoundException();
     }
-
-    public ProprietaireDTO getProprietaireByEmail(@NonNull String email){
-        Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findFirstByEmail(email);
-        if (optionalProprietaire.isEmpty()) {
-            throw new ProprietaireNotFoundException();
-        }
-
-        return ProprietaireMapper.entityToDto(optionalProprietaire.get(), false);
+    return ProprietaireMapper.entityToDto(optionalProprietaire.get(), isLogement);
+  }
+
+  /**
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.